python/ml4ir/base/model/calibrations/temperature_scaling.py

```diff
@@ -0,0 +1,247 @@
+    shutil.make_archive(zip_dir_path, "zip", root_dir, TEMPERATURE_SCALE)
+
+    # removing the dir, keeping only the zip
+    shutil.rmtree(zip_dir_path)
+    return zip_dir_path
+
+
+def get_intermediate_model(model, scorer):
+    """
+    returns a tf.keras.models.Model copy of the relavence_model.model. This model must generate
+    logits (inputs of softmax).
+    Parameters:
+        model:
+        scorer:
+    """
+    # get  last layer's output  --> MUST **NOT** BE AN ACTIVATION (e.g. SOFTMAX) LAYER
+    final_layer_name = scorer.model_config['layers'][-1]['name']
+    layer_output = model.get_layer(name=final_layer_name).output
+
+    return tf.keras.models.Model(inputs=model.input, outputs=layer_output)
+
+
+def eval_relevance_model(scorer: ScorerBase, logits, labels, temperature=None):
+    """
+    evaluate the relevance model given the logits and labels
+    Parameters:
+        scorer: Scorer of the RelevanceModel
+        logits: numpy array, logits (input of softmax)
+        labels: numpy array, class labels
+        temperature: TF.Tensor, a parameter of size=(1)
+    Returns:
+        accuracy and NLL loss
+    """
+    # convert to TF.Tensor
+    logits_tensor = tf.constant(logits, name='logits_test', dtype=tf.float32)
+    labels_tensor = tf.constant(labels, name='labels_test', dtype=tf.int32)
+
+    if temperature:
+        logits_tensor = tf.divide(logits_tensor, temperature)
+    nll = tf.reduce_mean(
+        tf.nn.sparse_softmax_cross_entropy_with_logits(labels=labels_tensor, logits=logits_tensor))
+
+    # getting the softmaxes using the model final activation function
+    softmaxes = scorer.final_activation_op(logits_tensor, {})
+    return accuracy(softmaxes, labels_tensor), tf.identity(nll), softmaxes
+
+
+def get_logits_labels(model: tf.keras.Model, evaluation_set: tf.data.TFRecordDataset):
+    """
+    predicts model output on the given evaluation set
+    Returns:
+        model output: numpy.array
+        labels: numpy.array
+    """
+    logits_nps = model.predict(evaluation_set).squeeze()
+    labels_nps = np.concatenate(list(map(lambda x: x[1], evaluation_set))).squeeze().argmax(axis=-1)
+    return logits_nps, labels_nps
+
+
+def temperature_scale(model: tf.keras.Model, scorer: ScorerBase, dataset: RelevanceDataset,
+                      logger: Logger,
+                      logs_dir_local: str, temperature_init: float):
+    """
+    learns a temperature parameter using Temperature Scaling (TS) technique on the validation set.
+    It, then, computes the probability scores of the test set with and without TS and writes them
+    in a .zip file.
+    Parameters:
+        model : tf.keras.Model object to be used for temperature scaling
+        scorer:  ScorerBase object of the RelevanceModel
+        dataset: RelevanceDataset to be used for temperature scaling.
+        logger : Logger
+        logs_dir_local: path to save the TS scores
+        temperature_init: temperature initial value
```

Comment:
@marjanhs Also add what you are returning. Check the format [here](https://numpydoc.readthedocs.io/en/latest/format.html)

Reply:
this function returns nothing